funding/manager_test.go

```diff
@@ -3803,3 +3805,44 @@ func TestFundingManagerFundingDoubleSpend(t *testing.T) {
 
 	}
 }
+
+// TestFundingManagerFundingDoubleSpend tests that we correctly handles the
+// case where we fail to broadcast the funding transaction, and/or another
+// conflicting transaction gets mined.
+func TestFundingManagerFundingDoubleSpendRestartBehaviour(t *testing.T) {
+	t.Parallel()
+
+	alice, bob := setupFundingManagers(t)
+	defer tearDownFundingManagers(t, alice, bob)
+
+	// We will consume the channel updates as we go, so no
+	// buffering is needed.
+	updateChan := make(chan *lnrpc.OpenStatusUpdate)
+
+	// Run through the process of opening the channel, up until the
+	// funding transaction is broadcasted.
+	openChannel(t, alice, bob, 500000, 0, 1,
+		updateChan, true)
+
+	recreateAliceFundingManager(t, alice)
+
+	spenderHash, err := chainhash.NewHashFromStr("abc")
+	if err != nil {
+		t.Fatalf("unable to create hash: %v", err)
+	}
+
+	alice.mockNotifier.spendChannel <- &chainntnfs.SpendDetail{
+		SpenderTxHash: spenderHash,
+	}
+
+	spenderHash, err = chainhash.NewHashFromStr("abc")
+	if err != nil {
+		t.Fatalf("unable to create hash: %v", err)
+	}
+
+	alice.mockNotifier.spendChannel <- &chainntnfs.SpendDetail{
+		SpenderTxHash: spenderHash,
+	}
+
+	assertErrorSent(t, alice.msgChan)
```

Comment:
Same comment here re asserting an update in the state of the funding manager itself.